### ISSUE TITLE
"arrow-body-style" rule for returning objects with computed keys

### ISSUE THREAD

**Issue by NameFILIP** [eslint/eslint#5819]
**What version of ESLint are you using?**
2.7.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel with webpack

**Please show your full configuration:**
I use aibnb config

**What did you do? Please include the actual source code causing the issue.**
```
function dotNotationToObject(keyString, val) {
  return keyString.split('.').reduceRight((prev, curr) => {
    return {
      [curr]: prev,
    };
  }, val);
}

// Usage
dotNotationToObject('content.button.text', 'Example text');
```
**What did you expect to happen?**
No errors or warnings

**What actually happened? Please include the actual, raw output from ESLint.**
Error: "arrow-body-style" - Unexpected block statement surrounding arrow body.

**Comment by platinumazure** [comment]
Confirmed in online demo with default parser (espree). However, I believe this is working as designed-- you should be able to use a concise body:

```js
function dotNotationToObject(keyString, val) {
  return keyString.split('.').reduceRight((prev, curr) => ({
    [curr]: prev,
  }), val);
}
```

That works on the online demo.

**Comment by NameFILIP** [comment]
Elegant! I confirm that it works. Thanks a lot!

**Comment by platinumazure** [comment]
No problem. The ECMAScript grammar handles the ambiguity of an open brace after an arrow by assuming it must be a block, but you can force an expression with parentheses.

Please feel free to close this if you are happy with the solution and don't believe there is a bugfix or enhancement needed here. Thanks!

**Comment by NameFILIP** [comment]
I am more than happy with the solution! Thank you for your help! :+1:

**Comment by gyandeeps** [comment]
I think the rule should be smart enough to detect the above use case and not throw a warning.

**Comment by michaelficarra** [comment]
@gyandeeps No, that's exactly what this rule is designed to report.

**Comment by platinumazure** [comment]
If we want to enhance the rule to allow the proposed style, perhaps we
should discuss in a separate issue. Thanks!
